### DISCUSSION TITLE
slingbox 500 having trouble/a little confused with windows 10 setup of slinger/server ports/forwarding

### DISCUSSION THREAD

**Discussion by jaybeebradley** [GerryDazoo/Slinger#274] [Q&A]
Update.  Got it working on my 500 with win 10 pc, tivo mini connected to slingbox  - and i viewed it on vlc on a mac.   Since I was a little lost on exact "how to do it step by step" details - I just had a couple things wrong.  (for others who are not  well versed in all of this similar to me).

1. port numbers - when I tried to load the stream in VLC - i was using the wrong ip address and port number - I was using those of   the slingbox - and instead needed to use the ip address/port - OF THE SLINGER SERVER PC AND PORT 8080  - in this address that you load as a network stream in vlc  - http://192.168.1.10:8080/slingbox   (insert your slinger server computer ip here).   I made slinger server pc and slingbox static reserved ip addresses in my router which helped.
2. I specified 8080 as the server port in my config.ini - and for my slingbox port for my config.ini -   i initially had it wrong - running the slingerserver.exe python executable showed me what my slingbox port was in the output - that I then put in my config.ini file. 
3. I do need port forwarding I think because I will use it out of my home network.  Will be testing that shortly.   I need to be able to get to my slinger server ip/port- so will 'forward' that - not the slingbox ip.  
4.  slingbox ip has all services blocked by my router to avoid sling doing a firmware update/bricking it.   

(edited) Hello - I think i am somewhat close - but still not understanding exactly what i'm doing and what is wrong.  and sorry for likely dumb questions - but I'm completely  lost on the port opening/forwarding  and how to do that?  (and now i realize there's a port for the slingbox and a port for the slinger server???)   And do i set the ports or are they in the box/on my router where are the ports from?  what do i use? 

** sorry and thank you!!!!   for anyone with a moment to get me straight - sorry I'm just not used to the python/router/port forwarding operations exactly other than some dabbling though I am a technical person and ready to learn ***

changed config.ini to give 5208 as port for the slingbox given at top and 8080 as server port to 'listen for' lower down in the file   but still vlc doesn't show anything and reports errors-
 "unable to open the mrl" regardless of what port i specify.

i am playing a show from my tivo through the slingbox - that i can see locally when i view the slingbox component output on a tv. 
I don't understand how to do the port forwarding/open a port/ redirect using the port.  

my netgear router says "port forwarding" with a radio button
then "service name" - and lists ftp, http, icuii...etc. all sorts of stuff including quake?
if i select http ? it then asks for "external and internal starting and ending port"  ?????
do i just type in start and end 8080 or 5208?  - is that internal/external?   and if you see below when i try to run it it lists a slingbox with port 5208.   is there a port for the box and the server and which is internal which is external and is it http and tcp/udp for protocol?

I just have no idea here?

here's vlc error messages, output of the slingbox.exe in python when i run it- and then my  config.ini -    I think i am confused about port and port forwarding - how to do that - as well as i have turned off all services to the ip address in my network so maybe that's blocking me internally too?   argh.     (how can i turn on debug mode so i can see what's happening?) 

thanks/sorry for this verbose plea for assistance.  :) 

VLC-

Your input can't be opened:

VLC is unable to open the MRL 'http://192.168.1.26:5208/slingbox'. Check the log for details.

Your input can't be opened:

VLC is unable to open the MRL 'http://192.168.1.26:8088/slingbox'. Check the log for details.

output of slingbox.exe running in python showing terminal output that pups up  (seems encouraging like at least it says "using slingbox" and "ok".  so i'm not bricked?)

Version : 3.08d Running on Windows-10-10.0.19044-SP0 pid= 16984
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.26', 5208)
('192.168.1.26', 5208) OK
8080 Using slingbox at  ('192.168.1.26', 5208)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off

here's my config.ini   **** (is there anything other than my password I should obscure so it's not posted on the internet for malicious actors?)
[SLINGBOX]
sbtype="350/500/M1/M2"
;sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=****i put admin password here - it's not user password is it? ****
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. Also removes requirement for the netifaces module  
ipaddress=192.168.1.26
port=5208
; Replace with your finderid, If and only if you need remote access and don't  
;have a static ip and you plan on using my service to access you server remotely.
; Please read the release notes. Most people don't need this.
;finderid=BAD95DAED609364C8204F2BC5FBDC363

;If you've got more than one slingbox set name. THe server will use this when generating
; logs to make it easier to see what's going on.
;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
;VideoBandwidth=2000

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, 1=composite 2=svideo 3=component
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't set this the code will use the last configired input
; I recommend setting this value. If not set correctly you'll often only see a black screen.
VideoSource = 1

[SERVER]
; local port number for the server to listen on for connections
port=8080
;maxstreams=10
enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work, But as a start make it
;the same as your configured VideoSource 0-3. See above.
code=1
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
include=remote.txt

**Comment by sp222222** [comment]
69 in fourth octet is your device running the slinGER server. 

port 5208 is what that server sees as the correct port to use. You will have to change the config.ini to match that port 5208. you cannot change that 5208 port. you CAN change the port listed later on under the config.ini to whatever you like. 

Best bet is to do all this with the internet off, b/c I fear you will brick this device before using the proper blocking rules on your router to prevent such bricking.....

**Reply by jaybeebradley** [reply]
thanks.  ok that 192.168.1.69 is now gone (thanks - yes that was the win 10 pc ip address i have the server on)  since i did now specify 5208 as the port and 8088 as the server port to listen on in my config.ini file.  

so i think my problem is the port forwarding.

where and how do i specify for the slingbox to be able to send and receive info - port 5208 or 8088 or a range and what protocol?

I have blocked all services on the slingbox ip address in my router (to block sling servers from sending a firmware update that will brick my 500) - but has that also blocked my ability to connect to it internally?

argh.  confused.

(cleaning up my first post a bit)

**Reply by sp222222** [reply]
internally will work fine. TCP is the protocol.  make a rule where local slinGER server IP is forwarding the 2nd port listed in config.ini you made up (8088). I won't hurt to put in another rule but with 5208 second go around. fire up the slinGER and it should stream. (make sure your source is ON)

**Reply by RandomMan719** [reply]
jaybeebradley,
   If you are just using it on your LAN, there should be no reason to port forward.

Slinger Server connects to the slingbox using slingbox ipaddress and slingbox port, and you connect your media player to the server using http://serveripaddress:serverport/slingbox.  If you are running the server on the same computer you are trying to stream to, you can use the loopback address, so to connect just put this in your mediaplayer http://127.0.0.1:8088/slingbox  and it should connect, or http://192.168.1.69:8088/slingbox

**Reply by jaybeebradley** [reply]
Thanks SP and Random - duh - I just had port confustion and ip address confusion - finally got it to work.  VLC needed my slinger ip/port - not slingbox ip/port.  thanks for the help!
